pulsar-client-cpp/lib/Commands.cc

```diff
@@ -137,6 +141,7 @@ PairSharedBuffer Commands::newSend(SharedBuffer& headers, BaseCommand& cmd,
         headers.setWriterIndex(writeIndex);
     }
 
+    cmd.release_send();
```

Comment:
Ok, here it's not allocating a new object each time. 
In any case, instead of releasing it (which means, it would need to allocate one for each message published), we should clear the protobuf object. So that it gets reset to default values, without destroying the internal objects (they are just cached).